src/window-manager.js

```diff
@@ -72,18 +68,8 @@ class WindowManager {
     // Handle when all windows are closed (UI quit)
     app.on('window-all-closed', async () => {
       if (!this.isQuitting && !this.shutdownInProgress) {
-        // On macOS, keep app running, on other platforms quit
-        // Clear saved state since user closed all windows (didn't quit)
-        this.isClearingState = true;
-        this.isQuitting = true;
-        this.shutdownInProgress = true;
-        this.finalSaveCompleted = true; // Prevent before-quit from trying to save
-        this.stopSaver();
-        await this.clearSavedState();
-
-        if (process.platform !== 'darwin') {
-          app.quit();
-        }
+        // Always perform graceful shutdown when last window closes
+        this.handleGracefulShutdown();
```

Comment:
Changing the `window-all-closed` handler to always call `handleGracefulShutdown()` makes the process quit on macOS as soon as the last window is closed. The expected browser behavior (and Electron’s default) is to keep the app alive on macOS so the dock/menu can reopen a window. Please revert this.